### PR TITLE
OGM-1054 Additional type descriptors

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1054

New type descriptors for Neo4j remote, I've added them in core as they might come handy in the future for other dialects.

### CODE REVIEW

File: core/src/main/java/org/hibernate/ogm/type/impl/ByteArrayStringType.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+	public static final ByteArrayStringType INSTANCE = new ByteArrayStringType();
+
+	public ByteArrayStringType() {
+		super( StringMappedGridTypeDescriptor.INSTANCE, ByteArrayTypeDescriptor.INSTANCE );
+	}
+
+	@Override
+	public String getName() {
+		return "Byte_array";
```
Comment (gsmet) [inline core/src/main/java/org/hibernate/ogm/type/impl/ByteArrayStringType.java]: Is the uppercase "Byte" expected? If so, I'm wondering if the other types should be consistent with this naming scheme. See ByteMappedType for instance.

Shouldn't we add the _as_string suffix to be consistent with PrimitiveByteArrayStringType?

File: core/src/main/java/org/hibernate/ogm/type/impl/BinaryMappedType.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import org.hibernate.MappingException;
+import org.hibernate.engine.spi.Mapping;
+import org.hibernate.ogm.type.descriptor.impl.ByteArrayMappedGridTypeDescriptor;
+import org.hibernate.type.descriptor.java.PrimitiveByteArrayTypeDescriptor;
+
+/**
+ * @author Davide D'Alto
+ */
+public class BinaryMappedType extends AbstractGenericBasicType<byte[]> {
```
Comment (gsmet) [inline core/src/main/java/org/hibernate/ogm/type/impl/BinaryMappedType.java]: Is there a reason why it's called Binary? I would have expected PrimaryByteArrayMappedType?

File: core/src/main/java/org/hibernate/ogm/type/impl/BinaryMappedType.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ */
+package org.hibernate.ogm.type.impl;
+
+import org.hibernate.MappingException;
+import org.hibernate.engine.spi.Mapping;
+import org.hibernate.ogm.type.descriptor.impl.ByteArrayMappedGridTypeDescriptor;
+import org.hibernate.type.descriptor.java.PrimitiveByteArrayTypeDescriptor;
+
+/**
+ * @author Davide D'Alto
```
Comment (gsmet) [inline core/src/main/java/org/hibernate/ogm/type/impl/BinaryMappedType.java]: Javadoc.

File: core/src/main/java/org/hibernate/ogm/type/impl/ByteArrayStringType.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ */
+package org.hibernate.ogm.type.impl;
+
+import org.hibernate.MappingException;
+import org.hibernate.engine.spi.Mapping;
+import org.hibernate.ogm.type.descriptor.impl.StringMappedGridTypeDescriptor;
+import org.hibernate.type.descriptor.java.ByteArrayTypeDescriptor;
+
+/**
+ * @author Davide D'Alto
```
Comment (gsmet) [inline core/src/main/java/org/hibernate/ogm/type/impl/ByteArrayStringType.java]: Javadoc.

File: core/src/main/java/org/hibernate/ogm/type/impl/ByteMappedType.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ */
+package org.hibernate.ogm.type.impl;
+
+import org.hibernate.MappingException;
+import org.hibernate.engine.spi.Mapping;
+import org.hibernate.ogm.type.descriptor.impl.ByteMappedGridTypeDescriptor;
+import org.hibernate.type.descriptor.java.ByteTypeDescriptor;
+
+/**
+ * @author Davide D'Alto
```
Comment (gsmet) [inline core/src/main/java/org/hibernate/ogm/type/impl/ByteMappedType.java]: Javadoc.

File: core/src/main/java/org/hibernate/ogm/type/impl/ByteArrayStringType.java (hunk 1 above)
Comment (DavideD) [inline core/src/main/java/org/hibernate/ogm/type/impl/ByteArrayStringType.java]: We don't really have a convention for these names, actually this property is not even  used. I'll update anyway to keep a bit of consistency

File: core/src/main/java/org/hibernate/ogm/type/impl/BinaryMappedType.java (hunk 2 above)
Comment (DavideD) [inline core/src/main/java/org/hibernate/ogm/type/impl/BinaryMappedType.java]: None really, just a name I've seen used while creating the classes. I will rename it

**Comment by DavideD** [comment]
Added some javadoc

**Comment by gsmet** [comment]
Merged! Thanks!

**Pull request merged**